packages/grid/x-grid/src/tests/columnHeaders.XGrid.test.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import {
+  createClientRenderStrictMode,
+  // @ts-expect-error need to migrate helpers to TypeScript
+  fireEvent,
+  // @ts-expect-error need to migrate helpers to TypeScript
+  screen,
+  // @ts-expect-error need to migrate helpers to TypeScript
+  waitFor,
+} from 'test/utils';
+import { expect } from 'chai';
+import { GRID_COLUMN_HEADER_SEPARATOR_RESIZABLE_CSS_CLASS, XGrid } from '@material-ui/x-grid';
+import { getColumnHeaderCell } from 'test/utils/helperFn';
+
+const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+describe('<XGrid /> - Column Headers', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps = {
+    autoHeight: isJSDOM,
+    disableColumnResize: false,
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+        foundationYear: 1964,
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+        foundationYear: 1949,
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+        foundationYear: 1948,
+      },
+    ],
+  };
+
+  describe('GridColumnHeaderMenu', () => {
+    it('should close the menu of a column when resizing this column', async () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <XGrid
+            {...baselineProps}
+            columns={[
+              { field: 'brand', resizable: true },
+              { field: 'foundationYear', resizable: true },
+            ]}
+          />
+        </div>,
+      );
+
+      const columnCell = getColumnHeaderCell(0);
+
+      const menuIconButton = columnCell.querySelector('button[aria-label="Menu"]');
+
+      fireEvent.click(menuIconButton);
+      await waitFor(() => expect(screen.queryByRole('menu')).not.to.equal(null));
+
+      const separator = columnCell.querySelector('.MuiDataGrid-iconSeparator');
+      fireEvent.mouseDown(separator);
+      await waitFor(() => expect(screen.queryByRole('menu')).to.equal(null));
```

Comment:
Trigger a `mouseup` to remove the added listeners. Then, you'll be able remove that `useEffect`. This comment applies to the other test case too.

```suggestion
      await waitFor(() => expect(screen.queryByRole('menu')).to.equal(null));
      fireEvent.mouseUp(separator);
```

Reply:
https://github.com/mui-org/material-ui-x/pull/1989#discussion_r661042972

It does work, but @oliviertassinari you not a huge fan

Reply:
We can leave a TODO note to remove the `mouseup` later. I think that changing the code to only make this test to pass is worse. In a real world scenario, we'll have the mouse up after the mouse down so the listeners will be removed.

Reply:
OK fine for me, I'll add the mouseup + TODO :+1:

Reply:
The mouse up do look better. It better isolate the origin of the issue.